bee-storage/bee-storage/src/access/fetch.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::backend::StorageBackend;
+
+/// `Fetch<K, V>` trait extends the `StorageBackend` with `fetch` operation for the (key: K, value: V pair);
+/// therefore, it should be explicitly implemented for the corresponding `StorageBackend`.
+#[async_trait::async_trait]
+pub trait Fetch<K, V>: StorageBackend {
```

Comment:
If we ever need two fetch operations which have the same input type and the same output type, but do have a different inner implemenation, is this possible to do?

Reply:
That's not possible per se. The solution is to have wrapper types to better express what you are trying to do.

Reply:
okey true, wrapper types would handle it
so it won't be a problem